cli/Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -1395,25 +1395,47 @@ extension PackageInfo.Target {
     /// The path used as base for all the relative paths of the package (e.g. sources, resources, headers)
     func basePath(packageFolder: AbsolutePath) async throws -> AbsolutePath {
         let fileSystem = FileSystem()
+
         if let path {
             return packageFolder.appending(try RelativePath(validating: path))
-        } else {
-            let predefinedDirectories: [String]
-            switch type {
-            case .test:
-                predefinedDirectories = PackageInfoMapper.predefinedTestDirectories
-            default:
-                predefinedDirectories = PackageInfoMapper.predefinedSourceDirectories
+        }
+
+        let predefinedDirectories = type == .test
+            ? PackageInfoMapper.predefinedTestDirectories
+            : PackageInfoMapper.predefinedSourceDirectories
+
+        for dir in predefinedDirectories {
+            // Standard layout: Sources/TargetName/
+            let standardPath = packageFolder.appending(components: [dir, name])
+            if try await fileSystem.exists(standardPath) {
+                return standardPath
             }
-            let firstMatchingPath = try await predefinedDirectories
-                .map { packageFolder.appending(components: [$0, name]) }
-                .concurrentFilter { try await fileSystem.exists($0) }
-                .first
-            guard let mainPath = firstMatchingPath else {
-                throw PackageInfoMapperError.defaultPathNotFound(packageFolder, name, predefinedDirectories)
+
+            // SE-0162 layout: source files directly in Sources/
+            // https://github.com/swiftlang/swift-evolution/blob/main/proposals/0162-package-manager-custom-target-layouts.md
+            let directPath = packageFolder.appending(component: dir)
+            if try await hasSourceFiles(in: directPath, fileSystem: fileSystem) {
+                return directPath
             }
-            return mainPath
         }
+
+        throw PackageInfoMapperError.defaultPathNotFound(packageFolder, name, predefinedDirectories)
+    }
+
+    /// Check if directory contains source files (for SE-0162 support)
+    private func hasSourceFiles(in directory: AbsolutePath, fileSystem: FileSystem) async throws -> Bool {
+        guard try await fileSystem.exists(directory) else { return false }
+
+        // SPM recognized source extensions
+        // https://github.com/swiftlang/swift-package-manager/blob/main/Sources/PackageModel/SupportedLanguageExtension.swift
+        let sourceExtensions = ["swift", "c", "cpp", "cc", "cxx", "m", "mm"]
+        for ext in sourceExtensions {
+            guard try await fileSystem
+                .glob(directory: directory, include: ["*.\(ext)"])
+                .first(where: { _ in true }) == nil
+            else { return true }
+        }
```

Comment:
instead of doing a glob per extension, it's better to change the `include` array to be an array of all the extensions. In fact, you can combine that in a single glob, see: https://github.com/tuist/tuist/blob/main/cli/Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift#L901